r/R/filesystem.R

```diff
@@ -280,6 +280,21 @@ SubTreeFileSystem$create <- function(base_path, base_fs) {
   shared_ptr(SubTreeFileSystem, xp)
 }
 
+#' Copy files between FileSystems
+#'
+#' @param src_fs The FileSystem from which files will be copied.
+#' @param src_paths The paths of files to be copied.
+#' @param dest_fs The FileSystem into which files will be copied.
+#' @param dest_paths Where the copied files should be placed.
+#' @param chunk_size The maximum size of block to read before flushing
+#' to the destination file. A larger chunk_size will use more memory while
+#' copying but may help accommodate high latency FileSystems.
+#' @param use_threads Whether to copy files in parallel.
```

Comment:
The R package is set up to use a global use-threads setting rather than exposing that in every function signature. So instead of exposing `use_threads` as a function argument, call `options_use_threads()` in L295. See `as.data.frame.RecordBatch` for an example.

Reply:
will do